pkg/registry/apis/provisioning/resources/dualwriter.go

```diff
@@ -18,13 +18,24 @@ import (
 	"github.com/grafana/grafana/pkg/apimachinery/utils"
 )
 
-// DualReadWriter is a wrapper around a repository that can read and write resources
+// DualReadWriter is a wrapper around a repository that can read from and write resources
+// into both the Git repository as well as in Grafana. It isn't a dual writer in the sense of what unistore handling calls dual writing.
+
+// Standard provisioning Authorizer has already run by the time DualReadWriter is called
+// for incoming requests from external actors. For internal requests, the authorization
+// checks that are performed here offer additional protection.
+
+// qq: why do we need to run the authorization checks again for external actors who have
+// already successfully authorized at the provisioning authorizer?
+
 // TODO: it does not support folders yet
 type DualReadWriter struct {
 	repo    repository.ReaderWriter
 	parser  Parser
 	folders *FolderManager
 	access  authlib.AccessChecker
+
+	onlyApiServer bool // Used to determine authz rules for standalone vs embedded apiserver
```

Comment:
we can't just use this flag to determine that, in embedded, onlyApiServer can be true if the custom.ini has

```
[provisioning]
disable_controllers = true
```